jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/DefaultHudsonClient.java

```diff
@@ -304,12 +303,24 @@ protected HttpHeaders createHeaders(final String userInfo) {
 
     protected String getLog(String buildUrl) {
         try {
-            String fullUrl = (new URL(URI.create(buildUrl).toURL(), "consoleText")).toString();
-            return makeRestCall(fullUrl).getBody();
+            return makeRestCall(joinURL(buildUrl, "consoleText")).getBody();
         } catch (MalformedURLException mfe) {
             LOG.error("malformed url for build log", mfe);
         }
 
         return "";
     }
+
+    // join a base url to another path or paths - this will handle trailing or non-trailing /'s
+    public static String joinURL(String base, String ... paths) throws MalformedURLException {
```

Comment:
i was thinking about creating a core-collector or something along those lines to consolidate all of these